### ISSUE TITLE
Update HACKING.txt and Pyramid's setup.py to use pip

### ISSUE THREAD

**Issue by stevepiercy** [Pylons/pyramid#2474]
In relation to #2104, Pyramid's `HACKING.txt` and `setup.py` should be updated to use pip for consistency. However this is outside the scope of #2104 because it's not "docs" and it needs special attention from developers who work on Pyramid's core and understand implications of such a change.

The three scaffold's `setup.py`s have been updated for the Pyramid 1.7 release already on [docs/easy-install-to-pip.2104](https://github.com/Pylons/pyramid/tree/docs/easy-install-to-pip.2104), and may serve as examples of the proposed changes.

**Comment by stevepiercy** [comment]
# Changes to `setup.py`

## Current
```python
testing_extras = tests_require + [
    'nose',
    'coverage',
    'virtualenv', # for scaffolding tests
    ]
...
      extras_require={
          'testing': testing_extras,
          'docs': docs_extras,
          },
      tests_require=tests_require,
      test_suite="pyramid.tests",
```

## Proposed

```python
tests_require = [
    'WebTest >= 1.3.1',  # py3 compat
    'pytest',  # includes virtualenv
    'pytest-cov',
    ]
...
      extras_require={
          'testing': tests_require,
          'docs': docs_extras,
          },
```

Would this break anything I might have overlooked?

**Comment by mmerickel** [comment]
That particular change is fine to make. Our tox file is already using pip as well. We should update our HACKING docs to recommend the similar workflow but I can do that myself later tonight.

**Comment by bertjwregeer** [comment]
If we remove `nose` we need to update our tox to start using py.test as well instead of nosetests.

**Comment by bertjwregeer** [comment]
This is the tox file I use for `pyramid_authsanity`: https://github.com/usingnamespace/pyramid_authsanity/blob/master/tox.ini

**Comment by mmerickel** [comment]
Removing nose in pyramid core should not part of these changes. I renamed the PR to reflect that. I didn't realize that was something @stevepiercy was trying to do in his original paste. I thought he was just merging the testing extras together into a single array.

**Comment by stevepiercy** [comment]
My intent was to copy what we did in the scaffolds to Pyramid's `setup.py`, specifically merge testing extras into a single array *and* use py.test instead of nose, as my proposal above reflects.

As I understand the situation, neither is desired by the core team at this time, so I reverted `setup.py` to its original state.

**Comment by stevepiercy** [comment]
Closed by #2492
